### ISSUE TITLE
How to debug when HMR stops working?

### ISSUE THREAD

**Issue by mohanzhang** [justin808/bootstrap-sass-loader#30]
The symptom I suddenly started experiencing is that the hot loader no longer picks up any changes I make to files referenced by `bootstrap-sass.config.js` (e.g. `mainSass`, `bootstrapCustomizations`, etc).

Suppose I have

`mainSass: './app/assets/stylesheets/_main.sass'`

Then when I save `_main.sass`, I see no updates in my webpack-dev-server log. The command I am using to run that is:

`webpack-dev-server --config ./webpack-dev.config.js --hot --progress --colors --history-api-fallback --content-base build`

I realize this isn't helpful at all as far as getting help identifying a specific problem, but I am more curious about what to check for when something like this stops working. So far, I have tried checking out at a previous point where I know for sure that the HMR was working, but when I go back in history and `npm install` at that checkout, it *still* doesn't work (weird, right?).

Other things I have checked/observed:
- `bootstrap-sass.config.js` is definitely getting loaded, since I have the correct and latest styling upon the dev server startup.
- I have not modified `bootstrap-sass.config.js` since the time it was working, so it's not a configuration issue
- refreshing the browser page after a CSS change does nothing, but restarting webpack-dev-server works (presumably because it rebuilds everything from scratch)
- It makes no difference whether I am on localhost:8080/ or localhost:8080/other/path - it still doesn't work (thus ruling out `--history-api-fallback`? It would be the only thing I can think of config-wise that I have changed recently?)

Basically, if you have any intuitions that come to mind, I would be very grateful to hear them. And thanks Justin for putting together this loader --- it has saved me (and I'm sure others) a lot of time. Much appreciated!

**Comment by justin808** [comment]
Thanks for the kind words @mohanzhang. If you know anybody interested in working with a cool, 100% remote team, please send them my way! We're on the beach in Maui and you've got a picture of windsurfer! 

Anyway, regarding the hot reload, there's a known issue in that dependencies of the top level used to not be picked up, and that was supposed to change soon. I sometimes use a shell command to touch one of the top level sass files. By top level, I mean one of the files directly picked up by webpack. 

However, if you touch the mainSass, that should get picked up.

I've got instructions on debugging this in the README.md. Did you try those?

**Comment by mohanzhang** [comment]
(No shaka emoji, unfortunately --- I will settle for :surfer:)

So I think I am experiencing a slightly worse problem than dependencies of the top level not being picked up; namely, it's that the top-level dependencies *themselves* are not being picked up. Like, I am hitting save on `mainSass` and touching `mainSass` and doing all sorts of things to it, but webpack-dev-server just sits there motionless. If I hit save or touch a js file for instance, that gets picked up just fine.


(EDIT)

I left the above text up there in case someone else finds this and what I describe above is what you're experiencing. As I was writing that, it occurred to me to check what my version of sass-loader was (inspired by @justin808's comment that implied that there are known issues with sass-loader at the moment that are supposed to get fixed soon). It turns out I was on 1.0.4, which was the latest version just 20 days ago. Since then, 2.0.1 came out 9 days ago. Updating sass-loader to the latest version fixed everything, and now hot reloading works again!

All told, it looks like bootstrap-sass-loader was just an innocent bystander to this problem, and was not at fault at all. But at least I got to fantasize about throwing sick moves on the north shore of Maui while I tried to figure it out!  (seriously, ask @justin808 about his backyard) Thanks Justin ;) haha!

**Comment by justin808** [comment]
@mohanzhang I hope I see you my pad http://www.sugarranchmaui.com some day! Regarding your issue, should we add something to the readme? Any other changes?

**Comment by mohanzhang** [comment]
@justin808 I'm actually not sure how I got into the bad state, so I don't know if there's anything I could say to prevent someone else from having the same problem.

If anyone out there sees this and also ran into this problem, mind posting how you got here? And if upgrading sass-loader was the solution for you as well?
